### PR TITLE
Fix/issues 5

### PR DESCRIPTION
**Pull request by alex-melnyk**

### CODE REVIEW

**Comment by IgorGanapolsky** [comment]
What is the description of this pull request, and how does it relate to "Modifications Applied" pull request?

**Comment by IgorGanapolsky** [comment]
Unable to launch app.  Crash:
> E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-1
    Process: com.igorganapolsky.vibratingwatchapp, PID: 3241
    java.lang.IllegalStateException: Room cannot verify the data integrity. Looks like you've changed schema but forgot to update the version number. You can simply fix this by increasing the version number.
        at android.arch.persistence.room.RoomOpenHelper.checkIdentity(RoomOpenHelper.java:135)
        at android.arch.persistence.room.RoomOpenHelper.onOpen(RoomOpenHelper.java:115)
        at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onOpen(FrameworkSQLiteOpenHelper.java:151)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:310)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:194)
        at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:96)
        at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:54)
        at android.arch.persistence.room.RoomDatabase.compileStatement(RoomDatabase.java:244)
        at android.arch.persistence.room.SharedSQLiteStatement.createNewStatement(SharedSQLiteStatement.java:65)
        at android.arch.persistence.room.SharedSQLiteStatement.getStmt(SharedSQLiteStatement.java:72)
        at android.arch.persistence.room.SharedSQLiteStatement.acquire(SharedSQLiteStatement.java:87)
        at com.igorganapolsky.vibratingwatchapp.domain.local.TimersDao_Impl.disableAll(TimersDao_Impl.java:155)
        at com.igorganapolsky.vibratingwatchapp.domain.WatchRepository.lambda$disableAll$5(WatchRepository.java:67)
        at com.igorganapolsky.vibratingwatchapp.domain.-$$Lambda$WatchRepository$UOke5DJ4yvgXckQg-vyJI_madU8.run(Unknown Source:2)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)

**Review by IgorGanapolsky — CHANGES_REQUESTED** [review]
.

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/VibratingWatchApp.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.igorganapolsky.vibratingwatchapp.domain.Repository;
+import com.igorganapolsky.vibratingwatchapp.domain.WatchRepository;
+import com.igorganapolsky.vibratingwatchapp.domain.local.TimersDatabase;
+import com.igorganapolsky.vibratingwatchapp.manager.timer.CountdownManager;
+import com.igorganapolsky.vibratingwatchapp.manager.timer.WatchCountdownManager;
+import com.igorganapolsky.vibratingwatchapp.manager.vibration.BeepManager;
+import com.igorganapolsky.vibratingwatchapp.manager.vibration.WatchBeepManager;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/VibratingWatchApp.java]: unused import

**Comment by IgorGanapolsky** [comment]
Explain difference between **core** package and **domain** package

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/core/util/ViewModelFactory.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.igorganapolsky.vibratingwatchapp.presentation.details.TimerDetailsViewModel;
+import com.igorganapolsky.vibratingwatchapp.presentation.main.TimerListViewModel;
+import com.igorganapolsky.vibratingwatchapp.presentation.settings.SetTimerViewModel;
+
+public class ViewModelFactory implements ViewModelProvider.Factory {
+
+    private static ViewModelFactory factory;
+
+    private Repository repository;
+    private CountdownManager countdownManager;
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/core/util/ViewModelFactory.java]: can be `final` vars

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/WatchRepository.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import com.igorganapolsky.vibratingwatchapp.domain.model.TimerModel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+public class WatchRepository implements Repository {
+
+    private TimersDatabase timerDb;
+    private ExecutorService executor;
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/WatchRepository.java]: can be `final` vars

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/local/TimersDao.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    void insert(TimerEntity entity);
+
+    @Update
+    void update(TimerEntity entity);
+
+    @Query("SELECT * FROM timers")
+    LiveData<List<TimerEntity>> getAll();
+
+    @Query("SELECT * FROM timers WHERE id = :id")
+    LiveData<TimerEntity> observeById(int id);
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/local/TimersDao.java]: Where is this method used?

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/BuzzSetup.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package com.igorganapolsky.vibratingwatchapp.domain.model;
+
+public class BuzzSetup {
+
+   public enum Type {SHORT, LONG}
+
+    private Type buzzType;
+    private int buzzCount;
+    private int buzzTime;
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/BuzzSetup.java]: can be `final` vars

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/CountData.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    public void setCurrentProgress(int currentProgress) {
+        this.currentProgress = currentProgress;
+    }
+
+    public void setCurrentTime(String currentTime) {
+        this.currentTime = currentTime;
+    }
+
+    public void setAnimationNeeded(boolean animationNeeded) {
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/CountData.java]: where is this method used?

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/TimerModel.java [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        model.setBuzzCount(1);
+        model.setBuzzTime(5);
+        return model;
+    }
+
+    public TimerModel() {
+        this(0, 0, 0);
+    }
+
+    public TimerModel(int hours, int minutes, int seconds) {
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/TimerModel.java]: can be `private`

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/manager/timer/CountdownManager.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    LiveData<TimerModel> observeActiveModel();
+
+    void setTickListener(TickListener listener);
+
+    TimerModel getActive();
+
+    int getActiveId();
+
+    boolean isHasMoreRepeats();
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/manager/timer/CountdownManager.java]: what is the purpose of this method?

**Comment by IgorGanapolsky** [comment]
Would it make sense to put **manager** and **vibration** packages in `domain` or `core` packages?  Or perhaps make them into separate gradle modules.  I'm trying to understand the architecture followed here...

**Comment by IgorGanapolsky** [comment]
Thank you for externalizing values into **strings.xml** and **dimens.xml** 👍

**Comment by IgorGanapolsky** [comment]
Let's improve the speed of launching these screens:
https://drive.google.com/open?id=1hAE9OoLJpF2tK_rFMofIkjJv2Y1hT02J

**Comment by IgorGanapolsky** [comment]
Unable to tap on the RecyclerView elements:
https://drive.google.com/open?id=1-68aY5WvOb2nuAuC90I3MvZFGBDzMg78

**Review by IgorGanapolsky — CHANGES_REQUESTED** [review]
Please address PR remarks

**Comment by IgorGanapolsky** [comment]
Looks like this **circle dot** doesn't accurately reflect the number:
device-2019-03-15-170132


Also, can we make this **circle dot** bigger?  It is difficult for the user to drag it now.

**Comment by IgorGanapolsky** [comment]
Request: is it possible to make swiping **left-to-right** on this screen easier?  Right now, the user cannot exit this screen unless they carefully swipe in the proper area:
https://drive.google.com/open?id=1khwpmK7XtqBKefkwXsKkMk_nWj-OW9WV

**Comment by IgorGanapolsky** [comment]
Pressing the stop button a running timer should reset it to the correct value, not **### 00:00:00**:
https://drive.google.com/open?id=1NuIJH9BXJ0sgW6M-8Xl8jSjda_LZMveg

**Comment by IgorGanapolsky** [comment]
Looks like the **reset** button completely doesn't work:
https://drive.google.com/open?id=1PdfXPEzejdQgkWnUN5b61XIqTU6E7F8s

**Comment by IgorGanapolsky** [comment]
Shouldn't there be indicators on this screen to tell the user which timer is running and which is not?
device-2019-03-15-173652

**Comment by IgorGanapolsky** [comment]
Another issue: when the timer buzzes, user is currently not brought to the screen to _see it_ or _stop it_.  Instead, the watch screen remains off, or on whatever other app is open at that time, and the watch is buzzing.

**Expected behavior:** just like any good timer app, a triggered timer should open the triggered timer screen for the user
**Actual behavior:** nothing happens other than the watch is buzzing

**Comment by IgorGanapolsky** [comment]
There is also a warning in Logcat:
> W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.igorganapolsky.vibratingwatchapp flg=0x4000010 (has extras) } to com.android.vending/com.google.android.finsky.packagemanager.impl.PackageMonitorReceiverImpl$RegisteredReceiver

**Comment by IgorGanapolsky** [comment]
Incorrect flow in the app.  Steps to reproduce:
1) Create two timers
2) Launch & start first timer from Timers List screen
3) Go back to Timers List screen
4) Open the second timer, edit its settings

**Expected result:**
you should be back on Timers List screen
**Actual result:**
You are on the first timer's activation/status screen.  After that I cannot even open the **second** timer at all (can only open **first** timer)!
https://drive.google.com/open?id=1hdqVZuqOFQukFNDb1yRWI5z7GHGPN6c9

**Comment by IgorGanapolsky** [comment]
The **stop** button doesn't work:
https://drive.google.com/open?id=1YsZp95Fu3Mt9BXjWpz4fnU1xX59ItAPG

**Comment by IgorGanapolsky** [comment]
Warnings in Logcat like this should be fixed:
>  W/ActivityManager: Duplicate finish request for ActivityRecord{4d90de9 u0 com.igorganapolsky.vibratingwatchapp/.presentation.main.TimerListActivity

**Comment by IgorGanapolsky** [comment]
This warning should be fixed:
>  W/ActivityManager: Activity pause timeout for ActivityRecord{d9d18a1 u0 com.igorganapolsky.vibratingwatchapp/.presentation.settings.SetTimerActivity

**Comment by IgorGanapolsky** [comment]
Why is Logcat showing this?
> I/Choreographer: Skipped 46 frames!  The application may be doing too much work on its main thread.

What is going on the main thread?

**Comment by IgorGanapolsky** [comment]
Another issue:
after editing a timer's config, the app goes back to config screen (instead of Timers List screen)

**Comment by IgorGanapolsky** [comment]
What the heck is going on here???
https://drive.google.com/open?id=1QXG_C2660u-9pK0GR9O3NG6LKwrxgtZw

**Comment by yuliiatereshchenko** [comment]
> Looks like this **circle dot** doesn't accurately reflect the number:
> device-2019-03-15-170132
> 
> Also, can we make this **circle dot** bigger? It is difficult for the user to drag it now.

1 the circle dot reflect the current number for H. the current H number is less than 1, that is why it reflect 0
2 yes,  but its an improvement

**Comment by yuliiatereshchenko** [comment]
> Looks like the **reset** button completely doesn't work:
> https://drive.google.com/open?id=1PdfXPEzejdQgkWnUN5b61XIqTU6E7F8s

1 please, could you specify which journey describes behaviour of this button?

**Comment by yuliiatereshchenko** [comment]
> Another issue: when the timer buzzes, user is currently not brought to the screen to _see it_ or _stop it_. Instead, the watch screen remains off, or on whatever other app is open at that time, and the watch is buzzing.
> 
> **Expected behavior:** just like any good timer app, a triggered timer should open the triggered timer screen for the user
> **Actual behavior:** nothing happens other than the watch is buzzing

1 please, could you specify which journey describes that it should be brought to the screen to _see it_ or _stop it_?

**Comment by yuliiatereshchenko** [comment]
> after editing a timer's config, the app goes back to config screen (instead of Timers List screen)

1 there is no information in Journey 13 where should be user redirected after configuration is done

File: app/build.gradle [hunk 9]
```diff
@@ -32,30 +30,39 @@ android {
...
-    implementation "com.android.support:design:$support_version"
-    implementation "com.android.support:support-annotations:$support_version"
+    // support
+    def support_version = "28.0.0"
+    implementation 'androidx.wear:wear:1.0.0'
+    implementation 'androidx.percentlayout:percentlayout:1.0.0'
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+    implementation 'androidx.recyclerview:recyclerview:1.0.0'
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'com.google.android.material:material:1.0.0'
```
Comment (IgorGanapolsky) [inline app/build.gradle]: ```what is this```

**Pull request merged**